vnext/ReactUWP/Views/KeyboardEventHandler.cpp

```diff
@@ -0,0 +1,402 @@
+      m_KeyUpRevoker = uiElement.KeyUp(winrt::auto_revoke, m_keyUpCallback);
+  }
+}
+
+void KeyboardEventHandler::unhook()
+{
+  m_KeyDownRevoker.revoke();
+  m_KeyUpRevoker.revoke();
+}
+
+PreviewKeyboardEventHandlerOnRoot::PreviewKeyboardEventHandlerOnRoot(const std::weak_ptr<IReactInstance>& reactInstance)
+  :PreviewKeyboardEventHandler(
+    std::bind(&PreviewKeyboardEventHandlerOnRoot::OnPreKeyDown, this, _1, _2),
+    std::bind(&PreviewKeyboardEventHandlerOnRoot::OnPreKeyUp, this, _1, _2)),
+  m_wkReactInstance(reactInstance)
+{
+}
+
+void PreviewKeyboardEventHandlerOnRoot::OnPreKeyDown(winrt::IInspectable const& sender, winrt::KeyRoutedEventArgs const&args)
+{
+  DispatchEventToJs("topKeyDown", args);
+}
+
+void PreviewKeyboardEventHandlerOnRoot::OnPreKeyUp(winrt::IInspectable const& sender, winrt::KeyRoutedEventArgs const&args)
+{
+  DispatchEventToJs("topKeyUp", args);
+}
+
+HandledKeyboardEventHandler::HandledKeyboardEventHandler()
+{
+}
+
+void HandledKeyboardEventHandler::UpdateHandledKeyboardEvents(string propertyName, folly::dynamic const& value)
+{
+  if (propertyName == "keyDownEvents") {
+    m_handledKeyDownKeyboardEvents = KeyboardHelper::FromJS(value);
+  }
+  else if (propertyName == "keyUpEvents")
+  {
+    m_handledKeyUpKeyboardEvents = KeyboardHelper::FromJS(value);
+  }
+}
+
+void HandledKeyboardEventHandler::hook(XamlView xamlView)
+{
+  unhook();
+
+  EnsureKeyboardEventHandler();
+  m_previewKeyboardEventHandler->hook(xamlView);
+  m_keyboardEventHandler->hook(xamlView);
+}
+
+void HandledKeyboardEventHandler::unhook()
+{
+  if (m_previewKeyboardEventHandler)
+    m_previewKeyboardEventHandler->unhook();
+  if (m_keyboardEventHandler)
+    m_keyboardEventHandler->unhook();
+}
+
+void HandledKeyboardEventHandler::EnsureKeyboardEventHandler()
+{
+  if (!m_previewKeyboardEventHandler)
+  {
+    m_previewKeyboardEventHandler = make_unique<PreviewKeyboardEventHandler>(
+      std::bind(&HandledKeyboardEventHandler::KeyboardEventHandledHandler, this, KeyboardEventPhase::PreviewKeyDown, _1, _2),
+      std::bind(&HandledKeyboardEventHandler::KeyboardEventHandledHandler, this, KeyboardEventPhase::PreviewKeyUp, _1, _2)
+      );
+  }
+
+  if (!m_keyboardEventHandler)
+  {
+    m_keyboardEventHandler = make_unique<KeyboardEventHandler>(
+      std::bind(&HandledKeyboardEventHandler::KeyboardEventHandledHandler, this, KeyboardEventPhase::KeyDown, _1, _2),
+      std::bind(&HandledKeyboardEventHandler::KeyboardEventHandledHandler, this, KeyboardEventPhase::KeyUp, _1, _2)
+      );
+  }
+}
+
+void HandledKeyboardEventHandler::KeyboardEventHandledHandler(KeyboardEventPhase phase, winrt::IInspectable const& sender, winrt::KeyRoutedEventArgs const& args)
+{
+  HandledEventPhase currentEventPhase = (phase == KeyboardEventPhase::PreviewKeyUp || phase == KeyboardEventPhase::PreviewKeyDown)
+    ? HandledEventPhase::CAPTURING : HandledEventPhase::BUBBLING;
+
+  auto ev = KeyboardHelper::CreateKeyboardEvent(currentEventPhase, args);
```

Comment:
>ev [](start = 7, length = 2)

I see you've used 'event' in at least one other place. I think we should use that instead of the abbreviation everywhere if possible.